include/OpenSpaceToolkit/Astrodynamics/NumericalSolver.hpp

```diff
@@ -0,0 +1,261 @@
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @brief                      Defines a numerical ODE solver that use the Boost Odeint libraries. This class will be moved into OSTk-math in the future.
+
+class NumericalSolver
+{
+
+    public:
+
+        enum class IntegrationStepperType
+        {
+            RungeKuttaCashKarp54,
+            RungeKuttaFehlberg78
+        } ;
+
+        enum class IntegrationLogType
+        {
+            NoLog,
+            LogConstant,
+            LogAdaptive
+        } ;
+
+        typedef std::vector<double> StateVector ; // Container used to hold the state vector
+        typedef std::function<void(const StateVector&, StateVector&, const double)> SystemOfEquationsWrapper ; // Function pointer type for returning dynamical equation's pointers
+
+        typedef runge_kutta_cash_karp54<NumericalSolver::StateVector> error_stepper_type_54 ;
+        typedef runge_kutta_fehlberg78<NumericalSolver::StateVector> error_stepper_type_78 ;
+
+        /// @brief              Constructor
+        ///
+        /// @code
+        ///                     NumericalSolver numericalSolver = { anIntegrationLogType, anIntegrationStepperType, aTimeStep, aRelativeTolerance, anAbsoluteTolerance } ;
+        /// @endcode
+        ///
+        /// @param              [in] anIntegrationLogType An enum indicating the amount of verbosity wanted to be logged during numerical integration
+        /// @param              [in] anIntegrationStepperType An enum indicating the type of numerical stepper used to perform integration
+        /// @param              [in] aTimeStep A number indicating the initial guess time step the numerical solver will take
+        /// @param              [in] aRelativeTolerance A number indicating the relative intergration tolerance
+        /// @param              [in] anAbsoluteTolerance A number indicating the absolute intergration tolerance
+
+                                NumericalSolver                             (   const   NumericalSolver::IntegrationLogType& anIntegrationLogType               = NumericalSolver::IntegrationLogType::NoLog,
+                                                                                const   NumericalSolver::IntegrationStepperType& anIntegrationStepperType       = NumericalSolver::IntegrationStepperType::RungeKuttaCashKarp54,
+                                                                                const   Real&                       aTimeStep                                   = 5.0,
+                                                                                const   Real&                       aRelativeTolerance                          = 1.0e-15,
+                                                                                const   Real&                       anAbsoluteTolerance                         = 1.0e-15   ) ;
```

Comment:
Don't put defaults in headers: this is too dangerous. Use `Real::Undefined()` instead.

Reply:
fixed